reconcile.go

```diff
@@ -518,6 +527,115 @@ func (r *TerraformReconciler) applyTerraformValueToCrdStatus(value *cty.Value, c
 	return nil
 }
 
+func (r *TerraformReconciler) getAttributeForCtyKey(key string, block *configschema.Block) (*configschema.Attribute, error) {
+	if block == nil {
+		return nil, fmt.Errorf("Cannot find attribute in nil block")
+	}
+
+	// Is the attribute in this block?
+	for k, v := range block.Attributes {
+		if k == key {
+			log.Printf("Key %s found", key)
+			return v, nil
+		}
+	}
+
+	// Is the attribute in a child block?
+	for bKey, bVal := range block.BlockTypes {
+		// Is the key a block?
+		if bKey == key {
+			log.Printf("Key %s is a nested block", bKey)
+			return nil, nil // nil, nil indicates this key belongs to a block not an attribute
+		}
+		a, e := r.getAttributeForCtyKey(bKey, &bVal.Block)
+		if a != nil {
+			return a, nil // We found an attribute
+		}
+		if a == nil && e == nil {
+			return nil, nil // We found a block
+		}
+		// An error here just indicates we couldn't find the attribute or child block in the current block - we should keep looking
+	}
+
+	return nil, fmt.Errorf("Unable to find attribute or block with the key %s", key)
+}
+
+func (r *TerraformReconciler) getValueFromCtyValue(key string, value *cty.Value, block *configschema.Block) (interface{}, error) {
```

Comment:
Nice work, fun times here - can we add a todo comment that in future would be nice to wrap some tests around the conversion? Happy to take it without as we have similar code in `create.go` which needs similar tests.

Reply:
Done